src/components/shared/Form/Onboarding/Steps/Steps.tsx

```diff
@@ -46,14 +52,59 @@ const Step1: React.FC<Step1Props> = ({
         type={"text"}
         placeholder="Full Name"
         register={register}
-        initialValue={session?.user?.name}
       />
       <Input
         label="portfolio"
         type={"url"}
         placeholder="Portfolio"
         register={register}
       />
+      <label
+        className="text-p3Med capitalize text-myWhite-300"
+        htmlFor={"location"}
+      >
+        {"Location"}
+      </label>
+      <Controller
+        name="location"
+        control={control}
+        render={({ field: { onChange, value, ref } }) => (
+          console.log(value),
+          (
+            <Autocomplete
+              style={{
+                //truncate
+                overflow: "hidden",
+                textOverflow: "ellipsis",
+                whiteSpace: "nowrap",
+                //rounded-sm
+                borderRadius: "0.125rem",
+                //border-none
+                borderStyle: "none",
+                //bg-myBlack-700
+                backgroundColor: "rgb(var(--my-black-700))",
+                //p-2
+                padding: "0.5rem",
+                //text-p4Reg
+                fontSize: "12px",
+                lineHeight: "16px",
+                fontWeight: 400,
+                //text-myWhite-300
+                color: "rgb(var(--my-white-300))",
+                //outline-none
+                outline: "2px solid transparent",
+                outlineOffset: "2px",
+              }}
+              apiKey={process.env.NEXT_PUBLIC_GMAPS}
+              onPlaceSelected={(place) => {
+                console.log(place);
+                onChange(place.formatted_address);
+              }}
+              defaultValue={value}
+            />
+          )
+        )}
```

Comment:
Regarding the error in your video:

## Address
Pressing "Enter" fires the 'onPlaceSelected' event with whatever the user typed instead of with a selection. If you look at the console.log you have for 'place':
```js
  onPlaceSelected={(place) => {
                console.log(place);
                onChange(place.formatted_address);
              }}
```
You can see this log appear:
![image](https://github.com/dalbanhi/git-note/assets/4108484/cb1f1bb1-21d1-4f58-bcb8-62bba8c47c0c)

We get an object without formatted_address, but instead just a `name` key.

The problem is that we're grabbing the "place.formatted_address", but not handling the case where the user typed a location manually.

We can address this by falling back to the "place.name" if "formatted_address" doesn't exist.

```js
  onChange(place?.formatted_address || place?.name);
 ```

## Enter Button
We also have an issue where pressing the `Enter` key inside this form element is submitting buttons. Because it's all in the same form- the Enter button is being used as a shortcut to "submit".

We can prevent the default behavior here by adding an "onKeyDown" event and, if the key is the 'Enter' key, stopping the default behavior:


```js
 <Autocomplete
         ...
              onKeyDown={(e) => {
                if (e.key === "Enter") {
                  e.preventDefault();
                  e.stopPropagation();
                }
              }}
 ```

Reply:
Very helpful, and that makes sense. I had only encountered stopPropagation a few times before, but this makes a lot of sense!